doomed_patient/doomed_patient/Player.cpp

```diff
@@ -5,11 +5,21 @@
 
 Player::Player(PatientGame* game, VectorXY startCoordinates)
 	: Character(game, startCoordinates, game->getPlayerSprite()),
-	speed(2)
+	speed(0)
 {
+	currentCell = game->level.getCell(VectorXY(0, 0));
+	centre = currentCell->getCentre();
 }
 
 void Player::move(Directions::Direction direction)
 {
-	centre += Directions::getDirectionVector(direction) * speed;
+	
+	// Put collision detection here
+
+	if (currentCell->getEdge(direction)->isWall())
+	{
+		return;
+	}
+	currentCell = game->level.getCell(currentCell->getCoordinates() + Directions::getDirectionVector(direction));
+	centre = currentCell->getCentre();
```

Comment:
The issue isn't with the keyboard checking, the issue is that the player is automatically being moved to the centre of the cell in the direction that's being pressed